app/src/main/java/com/geekbrains/onlinelessons/android/one/weatherapp/MainActivity.java

```diff
@@ -1,14 +1,244 @@
 package com.geekbrains.onlinelessons.android.one.weatherapp;
 
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
+import android.os.Handler;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.geekbrains.onlinelessons.android.one.weatherapp.wetherinternet.WeatherRequest;
+import com.google.gson.Gson;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import static com.geekbrains.onlinelessons.android.one.weatherapp.SettingForCity.settingKey;
 
 public class MainActivity extends AppCompatActivity {
+      private Button button;
+      private Button buttonGoogle;
+      private Button setting;
+      private static final String tag = "activity";
+      private ImageView imageView;
+      private TextView temp;
+      private TextView kPa;
+      private TextView wind;
+      private TextView wet;
+      private TextView city;
+      static String cityKey = "City";
+      private static final int requestCodeSet = 1234;
+      private static final int requestCodeSet2 = 12345;
+      private RecyclerView recyclerView;
+      public ArrayList<String> listForecast;
+      private String[] list;
+      private Button update;
+      private String cityUrl;
+
+    private static final String TAG = "WEATHER";
+   // private static final String WEATHER_URL = "https://api.openweathermap.org/data/2.5/weather?q=Moscow&appid=";
+   private static final String WEATHER_URL = "https://api.openweathermap.org/data/2.5/weather?q=";
+
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        Toast.makeText(this, "hello man",Toast.LENGTH_SHORT).show();
+        Log.d(tag,"Create");
+        initViews();
+        ButtonGoogle();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.